interface/modules/custom_modules/oe-module-faxsms/library/rc_sms_notification.php

```diff
@@ -0,0 +1,309 @@
+                    }
+                    $strMsg = "<b>* " . xlt("SEND NOTIFICATION BEFORE:") . text($SMS_NOTIFICATION_HOUR) . " | " . xlt("CRONJOB RUNS EVERY:") . text($CRON_TIME) . " | " . xlt("APPOINTMENT DATE TIME:") . ' ' . $app_date . " | " . xlt("APPOINTMENT REMAINING HOURS:") . " " . text($remaining_app_hour) . " | " . xlt("SEND ALERT AFTER:") . ' ' . text($remain_hour) . "</b>";
+
+                    // check in the interval
+                    if ($remain_hour >= -($CRON_TIME) && $remain_hour <= $CRON_TIME) {
+                        //set message
+                        $db_sms_msg['message'] = cron_SetMessage($prow, $db_sms_msg);
+                        $isValid = isValidPhone($prow['phone_cell']);
+                        // send sms to patient - if not in test mode
+                        if ($bTestRun == 0 && $isValid) {
+                            cron_InsertNotificationLogEntry($TYPE, $prow, $db_sms_msg);
+                            $clientApp->sendSMS(
+                                $prow['phone_cell'],
+                                $db_sms_msg['email_subject'],
+                                $db_sms_msg['message'],
+                                $db_sms_msg['email_sender']
+                            );
+                        }
+                        if (!$isValid) {
+                            $strMsg .= "<strong style='color:red'>\n* " . xlt("INVALID Mobile Phone#") . $prow['phone_cell'] . " " . xlt("SMS NOT SENT Patient:") . "</strong>" . text($prow['fname']) . " " . text($prow['lname']) . "</b>";
+                            $db_sms_msg['message'] = xlt("ERROR: INVALID Mobile Phone#") . text($prow['phone_cell']) . xlt("SMS NOT SENT For:") . " " . text($prow['fname']) . " " . text($prow['lname']);
+                            if ($bTestRun == 0) {
+                                cron_InsertNotificationLogEntry($TYPE, $prow, $db_sms_msg);
+                            }
+                        } else {
+                            $strMsg .= " | " . xlt("SENT SUCCESSFULLY TO") . "<strong> " . text($prow['phone_cell']) . "</strong>";
+                            cron_UpdateEntry($TYPE, $prow['pid'], $prow['pc_eid'], $prow['pc_recurrtype']);
+                        }
+                        if ((int)$prow['pc_recurrtype'] > 0) {
+                            $row = fetchRecurrences($prow['pid']);
+                            $strMsg .= "\n<b>" . xlt("A Recurring") . " " . text($row[0]['pc_catname']) . " " . xlt("Event Occuring") . " " . text($row[0]['pc_recurrspec']) . "</b>";
+                        }
+                        $strMsg .= "\n" . text($db_sms_msg['message']) . "\n";
+                        echo (nl2br($strMsg));
+                    }
+                }
+
+                unset($clientApp);
+                echo "<br /><h2>" . xlt("Done!") . "</h2>";
+                ?>
+            </div>
+        </body>
+    </html>
+<?php
+
+function isValidPhone($phone): array|bool|string|null
```

Comment:
woohooo on the php >= 8.0 union return types!

![dance-bear](https://user-images.githubusercontent.com/278968/221392336-3365ac3f-bc21-4cb2-830d-2a24d416c39e.gif)